packages/dev/core/src/AudioV2/abstractAudio/streamingSound.ts

```diff
@@ -0,0 +1,171 @@
+import { SoundState } from "../soundState";
+import type { ICommonSoundOptions, ICommonSoundPlayOptions } from "./abstractSound";
+import { AbstractSound } from "./abstractSound";
+import type { AudioEngineV2 } from "./audioEngineV2";
+import type { _StreamingSoundInstance } from "./streamingSoundInstance";
+
+/**
+ * Options for creating a streaming sound.
+ */
+export interface IStreamingSoundOptions extends ICommonSoundOptions {
+    /**
+     * The number of instances to preload. Defaults to 1.
+     * */
+    preloadCount: number;
+}
+
+/**
+ * Abstract class representing a streaming sound.
+ *
+ * A streaming sound has a sound buffer that is loaded into memory in chunks as it is played. This allows it to be played
+ * more quickly than a static sound, but it also means that it cannot have loop points or playback rate changes.
+ *
+ * Due to the way streaming sounds are typically implemented, there can be a significant delay when attempting to play
+ * a streaming sound for the first time. To prevent this delay, it is recommended to preload instances of the sound
+ * using the {@link IStreamingSoundOptions.preloadCount} options, or the {@link preloadInstance} and
+ * {@link preloadInstances} methods before calling the `play` method.
+ *
+ * Streaming sounds are created by the {@link CreateStreamingSoundAsync} function.
+ */
+export abstract class StreamingSound extends AbstractSound {
+    private _preloadedInstances = new Array<_StreamingSoundInstance>();
+    private _preloadedInstancesPromises = new Array<Promise<void>>();
+
+    protected override _options: IStreamingSoundOptions;
+
+    protected constructor(name: string, engine: AudioEngineV2, options: Partial<IStreamingSoundOptions> = {}) {
+        super(name, engine, options);
+
+        this._options = {
+            preloadCount: 1,
+            ...(this._options as ICommonSoundOptions),
+        };
+    }
+
+    /**
+     * The number of instances to preload.
+     */
+    public get preloadCount(): number {
+        return this._options.preloadCount;
+    }
+
+    /**
+     * Returns the number of instances that have been preloaded.
+     */
+    public get preloadedCount(): number {
+        return this._preloadedInstances.length;
+    }
```

Comment:
I get the difference, but the names are so close to each other I thought they were duplicates originally.